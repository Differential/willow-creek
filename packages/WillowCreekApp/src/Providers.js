import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
