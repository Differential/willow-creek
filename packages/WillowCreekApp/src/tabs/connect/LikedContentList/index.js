import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';

import { BackgroundView, FeedView } from '@apollosproject/ui-kit';

import getLikedContent from '../getLikedContent';
/** A FeedView wrapped in a query to pull content data. */
class LikedContentList extends PureComponent {
  /** Function for React Navigation to set information in the header. */
  static navigationOptions = () => ({
    title: 'Your Likes',
  });

  static propTypes = {
    /** Functions passed down from React Navigation to use in navigating to/from
     * items in the feed.
     */
    navigation: PropTypes.shape({
      getParam: PropTypes.func,
      navigate: PropTypes.func,
    }),
  };

  /** Function that is called when a card in the feed is pressed.
   * Takes the user to the ContentSingle
   */
  handleOnPress = (item) =>
    this.props.navigation.navigate('ContentSingle', {
      itemId: item.id,
      sharing: item.sharing,
    });

  render() {
    return (
      <BackgroundView>
        <Query query={getLikedContent} fetchPolicy="cache-and-network">
          {({ loading, error, data: { getAllLikedContent = [] }, refetch }) => (
            <FeedView
              initialNumToRender={5}
              content={getAllLikedContent}
              isLoading={loading}
              error={error}
              refetch={refetch}
              onPressItem={this.handleOnPress}
            />
          )}
        </Query>
      </BackgroundView>
    );
  }
}

export default LikedContentList;
