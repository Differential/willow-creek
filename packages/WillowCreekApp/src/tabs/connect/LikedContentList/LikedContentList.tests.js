import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from '../../../Providers';

import GET_LIKED_CONTENT from '../getLikedContent';
import LikedContentList from '.';

describe('LikedContentList component', () => {
  it('renders LikedContentList', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
      },
      result: {
        data: {
          likedContent: [
            {
              __typename: 'UniversalContentItem',
              id: 'UniversalContentItem:4148aeb9482181025e9cad42826e676a',
              title: 'Making Prayer a Spiritual Habit',
              coverImage: {
                sources: [
                  {
                    uri:
                      'https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31',
                  },
                ],
              },
              sharing: {
                title: 'Making Prayer a Spiritual Habit',
                message: '',
                url: 'https://apollosrock.newspring.cc/',
              },
            },
            {
              __typename: 'UniversalContentItem',
              id: 'UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e',
              title: 'Vacations and Your Budget',
              coverImage: {
                sources: [
                  {
                    uri:
                      'https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e',
                  },
                ],
              },
              sharing: {
                title: 'Vacations and Your Budget',
                message: '',
                url: 'https://apollosrock.newspring.cc/',
              },
            },
          ],
        },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LikedContentList navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
