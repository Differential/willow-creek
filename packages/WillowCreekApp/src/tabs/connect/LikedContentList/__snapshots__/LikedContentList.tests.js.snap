// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikedContentList component renders LikedContentList 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4294967295,
          4294506484,
        ]
      }
      locations={null}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "paddingVertical": 7.2,
          }
        }
        data={
          Array [
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                  },
                ],
              },
              "id": "UniversalContentItem:4148aeb9482181025e9cad42826e676a",
              "title": "Making Prayer a Spiritual Habit",
            },
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                  },
                ],
              },
              "id": "UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e",
              "title": "Vacations and Your Budget",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={5}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={0.7}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onPressItem={[Function]}
        onRefresh={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
          />
        }
        refreshing={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <RCTRefreshControl />
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 16,
                    "marginHorizontal": 16,
                    "marginVertical": 12,
                    "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 6,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderRadius": 16,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                            },
                          ]
                        }
                        style={
                          Object {
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "paddingHorizontal": 16,
                        "paddingVertical": 12,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Black",
                          "fontSize": 24,
                          "lineHeight": 27.6,
                        }
                      }
                    >
                      Making Prayer a Spiritual Habit
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "justifyContent": "space-between",
                        "paddingHorizontal": 16,
                        "paddingTop": 4,
                        "paddingVertical": 16,
                      }
                    }
                  >
                    <View
                      flexed={true}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "paddingRight": 8,
                          }
                        }
                        withIcon={null}
                      >
                        <Text
                          style={
                            Object {
                              "color": "#505050",
                              "fontFamily": "InterUI-Bold",
                              "fontSize": 12,
                              "lineHeight": 17.28,
                            }
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="19.2"
                      bbWidth="19.2"
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                          false,
                          Object {
                            "flex": 0,
                            "height": 19.2,
                            "width": 19.2,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                    >
                      <RNSVGPath
                        d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
                        fill={
                          Array [
                            0,
                            0.18823529411764706,
                            0.18823529411764706,
                            0.18823529411764706,
                            1,
                          ]
                        }
                        fillOpacity={1}
                        fillRule={1}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth="1"
                        x={0}
                        y={0}
                      />
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 16,
                    "marginHorizontal": 16,
                    "marginVertical": 12,
                    "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 6,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderRadius": 16,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                            },
                          ]
                        }
                        style={
                          Object {
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "paddingHorizontal": 16,
                        "paddingVertical": 12,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Black",
                          "fontSize": 24,
                          "lineHeight": 27.6,
                        }
                      }
                    >
                      Vacations and Your Budget
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "justifyContent": "space-between",
                        "paddingHorizontal": 16,
                        "paddingTop": 4,
                        "paddingVertical": 16,
                      }
                    }
                  >
                    <View
                      flexed={true}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "paddingRight": 8,
                          }
                        }
                        withIcon={null}
                      >
                        <Text
                          style={
                            Object {
                              "color": "#505050",
                              "fontFamily": "InterUI-Bold",
                              "fontSize": 12,
                              "lineHeight": 17.28,
                            }
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="19.2"
                      bbWidth="19.2"
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                          false,
                          Object {
                            "flex": 0,
                            "height": 19.2,
                            "width": 19.2,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                    >
                      <RNSVGPath
                        d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
                        fill={
                          Array [
                            0,
                            0.18823529411764706,
                            0.18823529411764706,
                            0.18823529411764706,
                            1,
                          ]
                        }
                        fillOpacity={1}
                        fillRule={1}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth="1"
                        x={0}
                        y={0}
                      />
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </BVLinearGradient>
  </View>
</RCTSafeAreaView>
`;
